source/common/network/ioctl_socket_option_impl.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "envoy/api/os_sys_calls.h"
+#include "envoy/common/platform.h"
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/network/listen_socket.h"
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Network {
+
+#define IOCTL_LEVEL 1
```

Comment:
Where did `1` come from? Is it windows specific?

Reply:
To comply with the `socket_option` interface in `#define ENVOY_MAKE_SOCKET_OPTION_NAME(level, option)` I need to set a level.

I can probably set it to `"IOCTL"` which will make it better for the reader

Reply:
I added a comment in 063cc63b664dde3d008b642bc53c0bcec7399d10 and renamed it to be look Envoy specific